src/Compilers/CSharp/Portable/Binder/Binder.WithQueryLambdaParametersBinder.cs

```diff
@@ -72,10 +72,12 @@ private BoundExpression SelectField(SimpleNameSyntax node, BoundExpression recei
                 {
                     // We only construct transparent query variables using anonymous types, so if we're trying to navigate through
                     // some other type, we must have some query API where the types don't match up as expected.
-                    // We should report this as an error of some sort.
-                    // TODO: DevDiv #737822 - reword error message and add test.
                     var info = new CSDiagnosticInfo(ErrorCode.ERR_UnsupportedTransparentIdentifierAccess, name, receiver.ExpressionSymbol ?? receiverType);
-                    Error(diagnostics, info, node);
+                    if (receiver.Type?.IsErrorType() != true)
```

Comment:
According to the language specification when we translate a Linq query such as `from x in e1 let y = e2 select M(x, y)` we create an anonymous type with a *transparent identifier* to represent both `x` and `y` in the final `select` clause. This can fall apart in (at least) two ways.

First, the `Select` method it translates into might not follow the *linq pattern`; it could, for example, be written to accept a `Func<int, int>` as its argument instead of a `Func<T, U>`. That would cause the compiler's normal translation "not to work" because the translation of `x` and `y` in the final `select clause` would just not work. This error message `ErrorCode.ERR_UnsupportedTransparentIdentifierAccess` is for that case, and tells the user that their Linq API may be malformed.

The second way it could fall apart is if there is some error earlier in the query expression. In that case the compiler would not be able to successfully do overload resolution, and would assign an `ErrorType` to the lambda argument. In this case there is no point telling the user that their API might be malformed; that would be an example of a *cascaded error message*. We always try to eliminate cascaded error messages, but we went too far before, simply dropping all diagnostics from lambdas we could not bind. Now we do better, but this particular diagnostic would be a cascaded error under the new scheme, so we suppress it when we have an error type.